hubbard/sp2.py

```diff
@@ -77,18 +77,21 @@ def sp2(ext_geom, t1=2.7, t2=0.2, t3=0.18, eB=3., eN=-3.,
     pi_geom.reduce()
 
     # Iterate over atomic species to set initial charge
-    for ia, (atom, _) in enumerate(pi_geom.atoms.iter(True)):
+    for atom, ia in pi_geom.atoms.iter(True):
 
         # Check if there are orbitals saved in the geometry
-        if not all(isinstance(orb, sisl.AtomicOrbital) for orb in atom.orbitals):
-            if orbitals is None:
-                # Default atomic orbitals is pz orbitals
-                pz = sisl.HydrogenicOrbital(2, 1, 0, 3.2, q0=atom.Z-5+dq, R=10.)
-
-            else:
-                pz = orbitals[ia]
-
+        if atoms is None:
+            pi_geom.atoms[ia] = sisl.Atom(atom.Z, no=1)
+            # Default atomic orbitals is pz orbitals
+            pz = sisl.HydrogenicOrbital(2, 1, 0, 3.2, q0=atom.Z-5+dq, R=10.)
```

Comment:
The way to do this is the other way around.

```python
pz =sisl.HydrogenicOrbital(2, 1, 0, 3.2, q0=atom.Z-5+dq, R=10.)
atom = sisl.Atom(atom.Z, pz)
```
then you get `Atom` initialization for free and correct stuff

Reply:
Thanks! Fixed this in commit [0233e96](https://github.com/dipc-cc/hubbard/pull/139/commits/0233e9674326d50787e7709e3b36f19ea0f6ee03). What do you think now?

Reply:
Since you are doing this for *all* C atoms, you should consider doing this instead:
https://zerothi.github.io/sisl/api/generated/sisl.Atoms.html#sisl.Atoms.replace_atom

Using the internal methods would be better since it ensures adjustments! :)

Reply:
Thanks @zerothi ! 

> Since you are doing this for _all_ C atoms, you should consider doing this instead: https://zerothi.github.io/sisl/api/generated/sisl.Atoms.html#sisl.Atoms.replace_atom
> 
> Using the internal methods would be better since it ensures adjustments! :)

Is something like what is now in commit [20cca0c](https://github.com/dipc-cc/hubbard/pull/139/commits/20cca0ce5caecb9a0c4bc8616ebabc4ef1e5f60a) what you meant?